e2e/tests-dfx/cycles-ledger.bash

```diff
@@ -248,6 +250,74 @@ current_time_nanoseconds() {
+  BOB=$(dfx identity get-principal --identity bob)
+  BOB_SUBACCT1="7C7B7A030405060708090a0b0c0d0e0f101112131415161718191a1b1c1d1e1f"
+
+  deploy_cycles_ledger
+
+  assert_command dfx canister call depositor deposit "(record {to = record{owner = principal \"$ALICE\";};cycles = 3_000_000_000_000;})" --identity cycle-giver
+  assert_command dfx canister call depositor deposit "(record {to = record{owner = principal \"$ALICE\"; subaccount = opt blob \"$ALICE_SUBACCT1_CANDID\"};cycles = 1_000_000_000_000;})" --identity cycle-giver
+
+  # account to account
+  assert_command dfx cycles balance --precise --identity alice
+  assert_eq "3000000000000 cycles."
+  assert_command dfx cycles balance --precise --identity bob
+  assert_eq "0 cycles."
+
+  t=$(current_time_nanoseconds)
+  assert_command dfx cycles approve "$BOB" 2000000000 --created-at-time "$t" --memo 123 --identity alice
+  assert_eq "Approval sent at block index 2"
+  assert_command dfx cycles approve "$BOB" 2000000000 --created-at-time "$t" --memo 123 --identity alice
+  assert_contains "Approval is a duplicate of block 2"
+  assert_command dfx cycles transfer "$BOB" 100000 --from "$ALICE" --identity bob
+  assert_eq "Transfer sent at block index 3"
+
+  assert_command dfx cycles balance --precise --identity alice
+  assert_eq "2999799900000 cycles."
+
+  assert_command dfx cycles balance --precise --identity bob
+  assert_eq "100000 cycles."
+
+  # account to subaccount
+  assert_command dfx cycles balance --precise --identity alice
+  assert_eq "2999799900000 cycles."
+  assert_command dfx cycles balance --precise --identity bob --subaccount "$BOB_SUBACCT1"
+  assert_eq "0 cycles."
+
+  t=$(current_time_nanoseconds)
+  assert_command dfx cycles transfer "$BOB" 100000 --from "$ALICE" --to-subaccount "$BOB_SUBACCT1" --created-at-time "$t" --identity bob
+  assert_eq "Transfer sent at block index 4"
+  assert_command dfx cycles transfer "$BOB" 100000 --from "$ALICE" --to-subaccount "$BOB_SUBACCT1" --created-at-time "$t" --identity bob
+  assert_contains "Transfer is a duplicate of block index 4"
+
+  assert_command dfx cycles balance --precise --identity alice
+  assert_eq "2999699800000 cycles."
+
+  assert_command dfx cycles balance --precise --identity bob --subaccount "$BOB_SUBACCT1"
+  assert_eq "100000 cycles."
+
+  # subaccount to account
+  assert_command dfx cycles balance --precise --identity alice --subaccount "$ALICE_SUBACCT1"
+  assert_eq "1000000000000 cycles."
+  assert_command dfx cycles balance --precise --identity bob
+  assert_eq "100000 cycles."
+
+  assert_command dfx cycles approve "$BOB" 200000000000 --from-subaccount "$ALICE_SUBACCT1" --identity alice
+  assert_eq "Approval sent at block index 5"
+  assert_command dfx cycles transfer "$BOB" 700000 --from "$ALICE" --from-subaccount "$ALICE_SUBACCT1" --identity bob
+  assert_eq "Transfer sent at block index 6"
+
```

Comment:
Is there a way to include the `--spender-allowance` and `--spender-subaccount` parameters in the test?